main/acle.md

```diff
@@ -9546,25 +9680,2141 @@ possible to write these operations using normal C arithmetic. For example:
+  //   _za32[_u32]
+  //   _za64[_f64] (only if __ARM_FEATURE_SME_F64F64 != 0)
+  //   _za64[_u64] (only if __ARM_FEATURE_SME_I16I64 != 0)
+  __attribute__((arm_streaming, arm_shared_za))
+  void svsub_za32[_f32]_vg1x2(uint32_t slice, svfloat32x2_t zm);
+  
+  
+  // Variants are available for:
+  //   _za32[_f32]
+  //   _za32[_u32]
+  //   _za64[_f64] (only if __ARM_FEATURE_SME_F64F64 != 0)
+  //   _za64[_u64] (only if __ARM_FEATURE_SME_I16I64 != 0)
+  __attribute__((arm_streaming, arm_shared_za))
+  void svsub_za32[_f32]_vg1x4(uint32_t slice, svfloat32x4_t zm);
+  
+  ```
+
+#### BFCVTN, FCVTN
+
+Multi-vector floating-point convert from single-precision to interleaved half-precision/BFloat16
+
+``` c
+  // Variants are also available for _f16[_f32_x2]
+  __attribute__((arm_streaming))
+  svbfloat16_t svcvtn_bf16[_f32_x2](svfloat32x2_t zn);
+  
+  ```
+
+#### FCVT, BFCVT, FCVTZS, FCVTZU, SCVTF, UCVTF
+
+Multi-vector convert to/from floating-point.
+
+``` c
+  // Variants are also available for _f16[_f32_x2]
+  __attribute__((arm_streaming))
+  svbfloat16_t svcvt_bf16[_f32_x2](svfloat32x2_t zn);
+
+
+  // Variants are also available for _f32[_u32_x2], _s32[_f32_x2] and _u32[_f32_x2]
+  __attribute__((arm_streaming))
+  svfloat32x2_t svcvt_f32[_s32_x2](svint32x2_t zn);
+  
+  
+  // Variants are also available for _f32[_u32_x4], _s32[_f32_x4] and _u32[_f32_x4]
+  __attribute__((arm_streaming))
+  svfloat32x4_t svcvt_f32[_s32_x4](svint32x4_t zn);
+  
+  ```
+
+#### SQCVT, SQCVTU, UQCVT
+
+Multi-vector saturating extract narrow
+
+``` c
+  // Variants are also available for _u16[_s32_x2] and _u16[_u32_x2]
+  __attribute__((arm_streaming))
+  svint16_t svqcvt_s16[_s32_x2](svint32x2_t zn);
+  
+  
+  // Variants are also available for _u8[_s32_x4], _u8[_u32_x4], _s16[_s64_x4],
+  // _u16[_s64_x4] and _u16[_u64_x4]
+  __attribute__((arm_streaming))
+  svint8_t svqcvt_s8[_s32_x4](svint32x4_t zn);
+  
+  ```
+
+#### SQCVTN, SQCVTUN, UQCVTN
+
+Multi-vector saturating extract narrow and interleave
+
+``` c
+  // Variants are also available for _u16[_s32_x2] and _u16[_u32_x2]
+  __attribute__((arm_streaming_compatible))
+  svint16_t svqcvtn_s16[_s32_x2](svint32x2_t zn);
+  
+  
+  // Variants are also available for _u8[_s32_x4], _u8[_u32_x4], _s16[_s64_x4],
+  // _u16[_s64_x4] and _u16[_u64_x4]
+  __attribute__((arm_streaming))
+  svint8_t svqcvtn_s8[_s32_x4](svint32x4_t zn);
+  
+  ```
+
+#### UDOT, SDOT, FDOT (vectors)
+
+Multi-vector dot-product (2-way and 4-way)
+
+``` c
+  // Variants are also available for _s32_s16_s16 and _u32_u16_u16
+  __attribute__((arm_streaming_compatible))
+  svfloat32_t svdot[_f32_f16_f16](svfloat32_t zda, svfloat16_t zn, svfloat16_t zm);
+  
+  ```
+
+#### UDOT, SDOT, FDOT (indexed)
+
+Multi-vector dot-product (2-way and 4-way)
+
+``` c
+  // Variants are also available for _s32 and _u32
+  __attribute__((arm_streaming_compatible))
+  svfloat32_t svdot_lane[_f32](svfloat32_t zda, svfloat16_t zn, svfloat16_t zm,
```

Comment:
There are already 4-way functions with the names `svdot_lane[_s32]` and `svdot_lane[_u32]`, so the new ones will need to have an extra suffix, like for the vector-vector version above.  In future, we'll also need a 4-way version of FDOT for:

* https://developer.arm.com/documentation/ddi0602/2023-09/SIMD-FP-Instructions/FDOT--4-way--by-element---8-bit-floating-point-dot-product-to-single-precision--vector--by-element--?lang=en

So I think we'll need an extra suffix for the `f32` case too.